config.cc

```diff
@@ -62,6 +71,19 @@ namespace
     return result;
   }
 
+  QDir getDataDir()
+  {
+#if QT_VERSION >= QT_VERSION_CHECK( 5, 0, 0 )
```

Comment:
Even better: if this function is called only in case of `HAVE_X11` and Qt 5, it can be enclosed in such `#if`. Then the `return getHomeDir();` fallback can be removed.

Reply:
It could. And yes, it is used only on Unixes for now. Personally, I think it's not that important, because compiler would remove dead code on other platforms, but since you prefer using preprocessor, then I will change that.

Reply:
Dead code is no good: could cause compiler warnings. In this case using preprocessor will make it clearer that this function is limited to one platform and newer Qt.

Reply:
Preprocessor directives are also problematic. Tools e.g. autocompletion often have problem to understand it. Anyway, change that, but please check if that's what you had in mind.

Reply:
> Preprocessor directives are also problematic. Tools e.g. autocompletion often have problem to understand it. 

True. Unfortunately, supporting many platforms and ancient Qt versions tends to create such preprocessor mazes.

> Anyway, change that, but please check if that's what you had in mind.

Yes, that's it. Thanks.